### PR TITLE
Add more Flow annotations

### PR DESCRIPTION
**Pull request by lencioni**
While working on this, I ran into what looks like a bug to me in
Flow.

  https://github.com/facebook/flow/issues/1789

I modified Configuration#constructor to work around this for now.

### CODE REVIEW

File: lib/ImportStatements.js [hunk 1]
```diff
@@ -1,9 +1,14 @@
+// @flow
+
 // Class that sorts ImportStatements as they are pushed in
 import flattenDeep from 'lodash.flattendeep';
 import partition from 'lodash.partition';
 import sortBy from 'lodash.sortby';
 import uniqBy from 'lodash.uniqby';
 
+import Configuration from './Configuration';
+import ImportStatement from './ImportStatement';
```
Comment (trotzig) [inline lib/ImportStatements.js]: I like how this makes dependencies more obvious.

File: lib/Importer.js [hunk 2]
```diff
@@ -329,62 +310,55 @@ export default class Importer {
...
+        matchingImportStatement.namedImports.indexOf(variableName) !== -1;
+    }
+
+    const matchedModule = new JsModule({
+      importPath: matchingImportStatement.path,
+      hasNamedExports,
+    });
+
+    return matchedModule;
   }
```
Comment (trotzig) [inline lib/Importer.js]: Thanks for making this method better. It was a little hard to follow along with the diff though.

File: lib/ImportStatements.js (hunk 1 above)
Comment (lencioni) [inline lib/ImportStatements.js]: Yeah, me too!

File: lib/Importer.js (hunk 2 above)
Comment (lencioni) [inline lib/Importer.js]: Yeah, I should have broken that up into separate commits.
